experimental/devices/bitbang/spi.go

```diff
@@ -84,6 +87,15 @@ func (s *SPI) Connect(f physic.Frequency, mode spi.Mode, bits int) (spi.Conn, er
 	}
 	s.spiConn.mode = mode
 	s.spiConn.bits = bits
+	s.spiConn.readAfterClockPulse = mode&spi.Mode1 == spi.Mode1
+
+	// Set clock idle polarity
+	if mode&spi.Mode2 == spi.Mode2 {
```

Comment:
It's a bit of a hack but you can do:

    s.spiConn.clockIdle = gpio.Level(mode&spi.Mode2 == spi.Mode2)

Also, we would want to have the pin setup to idle state right here.

Reply:
I wouldn't normally compress the logic so much, but since the rest of the connect options are handled similarly, that sounds reasonable.